File: src/OB.Data/Entities/Vote.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace OB.Data.Entities
{
    public class Vote
    {
        [Key]
        public int Id { get; set; }

        [Required, StringLength(24)]
        public string Code { get; set; }                       // private identifier

        public int RegistrantId { get; set; }

        public byte[] Ballot { get; set; }

        public bool BallotCast { get; set; }

        // ----------------------------------------------  audit
        public DateTime DateCreated { get; set; }
    }
}

File: _assets/output_gen/jsModels/Vote.js

//---------------------------------------------------------------------------

export default class Vote {
  constructor(options) {

    this.Id = null;
    this.Code = null;
    this.RegistrantId = null;
    this.Ballot = null;
    this.BallotCast = null;
    this.DateCreated = null;

    if(options) {
      this.init(options);
    }
  }

  //---------------------------------------------------------------------------
  init(options) {

    this.Id = options.Id || this.Id;
    this.Code = options.Code || this.Code;
    this.RegistrantId = options.RegistrantId || this.RegistrantId;
    this.Ballot = options.Ballot || this.Ballot;
    this.BallotCast = options.BallotCast || this.BallotCast;
    this.DateCreated = options.DateCreated || this.DateCreated;

  }  
  //---------------------------------------------------------------------------  
}